src/metabase/automagic_dashboards/adaptive_dashboard_size.clj

```diff
@@ -0,0 +1,43 @@
+(ns metabase.automagic-dashboards.adaptive-dashboard-size
+  (:require [bigml.histogram.core :as hist]
+            [clojure.core.memoize :as memo]
+            [metabase.models
+             [database :refer [Database]]
+             [query-execution :refer [QueryExecution]]]
+            [metabase.sync.analyze.fingerprint.fingerprinters :refer [histogram]]
+            [redux.core :as redux]
+            [toucan.db :as db]))
+
+(def ^:private ^Long long-running-90th-percentile-threshold
+  "If 10% of queries take longer than this to run, consider the DB to be slow. Unit is ms."
+  5000)
+
+(def ^:private ^Long running-time-cache-ttl (* 60 60 24 7 1000)) ; 1 week in ms
+
+(def ^:private ^Long max-cards 15)
+(def ^:private ^Long max-cards-if-no-summary 3)
+
+(def ^:private ^{:arglists '([db-id])} running-time-90th-percentile
+  (memo/ttl #(transduce identity
+                        (redux/post-complete
+                         histogram
+                         (fn [h]
+                           ((hist/percentiles h 0.9) 0.9)))
+                        (db/select-field :running_time QueryExecution :database_id %))
+            :ttl/threshold running-time-cache-ttl))
+
+(defn- slow-db?
+  [db-id]
+  (some-> db-id running-time-90th-percentile (> long-running-90th-percentile-threshold)))
```

Comment:
we should use the same db setting that determines whether or not to automatically run queries in view mode, not replicate it in a subtly different way here.

Reply:
It uses both, the flag (line 41) and a heuristic